### PR TITLE
DEX-1508 fixed command ordering for AddressActor

### PR DESCRIPTION
**Pull request by JennaWestenra**

### CODE REVIEW

File: dex/src/main/scala/com/wavesplatform/dex/actors/events/OrderEventsCoordinatorActor.scala [hunk 1]
```diff
@@ -26,6 +26,16 @@ import play.api.libs.json.Json
 
 object OrderEventsCoordinatorActor {
 
+  final private case class SortedEvents(
+    added: Vector[AddressActor.Command.ApplyOrderBookAdded] = Vector.empty,
+    executed: Vector[AddressActor.OrderBookExecutedEvent] = Vector.empty,
+    cancelled: Vector[AddressActor.Command.ApplyOrderBookCanceled] = Vector.empty
+  ) {
+    def add(event: AddressActor.Command.ApplyOrderBookAdded): SortedEvents = copy(added = added :+ event)
```
Comment (alexander-branevskiy) [inline dex/src/main/scala/com/wavesplatform/dex/actors/events/OrderEventsCoordinatorActor.scala:34]: probably we can name all 3 ```add``` and rely on overloading

**Review by alxndrmkd — APPROVED** [review]

File: dex/src/test/scala/com/wavesplatform/dex/actors/orderbook/OrderBookActorSpecification.scala [hunk 2]
```diff
@@ -466,10 +468,14 @@ class OrderBookActorSpecification
         val taker1 = buy(wavesUsdPair, 10.waves, 3.00, matcherFee = 0.005.waves.some, ts = now.some)
         orderBook ! wrapLimitOrder(1, taker1)
         val events = tp.expectMsgType[Process].events
+        events.size shouldBe 2
         events.head shouldBe a[OrderAdded]
-        val oe = events.collect { case a: OrderExecuted => a }.head
-        oe.counterExecutedFee shouldBe 0.0006.waves
-        oe.submittedExecutedFee shouldBe 0.005.waves
+        events.last match {
```
Comment (alexander-branevskiy) [inline dex/src/test/scala/com/wavesplatform/dex/actors/orderbook/OrderBookActorSpecification.scala:473]: just use asInstanceOf

File: dex/src/test/scala/com/wavesplatform/dex/actors/orderbook/OrderBookActorSpecification.scala [hunk 3]
```diff
@@ -509,21 +515,30 @@ class OrderBookActorSpecification
...
         orderBook ! wrapMarketOrder(marketOrder)
-        val events = tp.expectMsgType[Process].events
+        val events = tp.expectMsgType[Process].events.toList
+        events.size shouldBe 4
         events.head shouldBe a[OrderAdded]
-        val oe1 = events.collect { case a: OrderExecuted => a }.head
-        oe1.submitted shouldBe marketOrder
-        oe1.counter shouldBe LimitOrder(counterOrder1)
-        oe1.executedAmount shouldBe counterOrder1.amount
+        val oe1 = events(1) match {
```
Comment (alexander-branevskiy) [inline dex/src/test/scala/com/wavesplatform/dex/actors/orderbook/OrderBookActorSpecification.scala:521]: asInstanceOf

File: dex/src/test/scala/com/wavesplatform/dex/actors/orderbook/OrderBookActorSpecification.scala [hunk 4]
```diff
@@ -588,13 +596,14 @@ class OrderBookActorSpecification
 
           orderBook ! wrapMarketOrder(marketOrder)
           val events = tp.expectMsgType[Process].events
+          events.size shouldBe 3
           events.head shouldBe a[OrderAdded]
           val oe = events.collect { case a: OrderExecuted => a }.last
```
Comment (alexander-branevskiy) [inline dex/src/test/scala/com/wavesplatform/dex/actors/orderbook/OrderBookActorSpecification.scala:601]: discussed

**Review by alexander-branevskiy — APPROVED** [review]

**Pull request merged**